### PR TITLE
style(EMS-3172): No PDF - Update/align various error messages

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
This PR updates various error messages to be aligned with the latest content iterations.

Also updates the related validation rules themselves to use DRY validation functions and `MAXIMUM` definitions from constants.

## Resolution :heavy_check_mark:
- Update `MAXIMUM_CHARACTERS` constants.
- Update error messages content strings with the latest content and `MAXIMUM_CHARACTERS` definitions.
- Remove the related `MAXIMUM` definitions from field content strings.
- Update E2E tests to use `MAXIMUM_CHARACTERS` definitions.
- Update some validation rules to:
  - Use `providedAndMaxLength`, instead of running the same duplicated logic.
  - Use `MAXIMUM_CHARACTERS` definitions.

## Miscellaneous :heavy_plus_sign:
- Update `MAXIMUM_CHARACTERS` constants to be ordered alphabetically.
- Fix some typos.

### CODE REVIEW

File: src/ui/server/controllers/insurance/business/nature-of-business/validation/rules/goods-or-services.ts [hunk 1]
```diff
@@ -1,36 +1,24 @@
...
-  }
-
-  if (formBody[FIELD_ID].length > MAXIMUM) {
-    return generateValidationErrors(FIELD_ID, ERROR_MESSAGE.ABOVE_MAXIMUM, errors);
-  }
-
-  return errors;
-};
+const goodsOrServices = (formBody: RequestBody, errors: object) =>
+  providedAndMaxLength(formBody, FIELD_ID, ERROR_MESSAGES_OBJECT, errors, MAXIMUM_CHARACTERS.NATURE_OF_BUSINESS_GOODS_OR_SERVICES_DESCRIPTION);
```
Comment (Zainzzkk) [inline src/ui/server/controllers/insurance/business/nature-of-business/validation/rules/goods-or-services.ts:22]: could NATURE_OF_BUSINESS_GOODS_OR_SERVICES_DESCRIPTION be dereferenced? it is quite long.  Not a blocker

File: src/ui/server/controllers/insurance/your-buyer/company-or-organisation/validation/rules/registration-number.ts [hunk 2]
```diff
@@ -28,7 +26,7 @@ export const MAXIMUM = Number(FIELDS.COMPANY_OR_ORGANISATION[FIELD_ID].MAXIMUM);
  */
 const registrationNumber = (formBody: RequestBody, errors: object) => {
   if (objectHasProperty(formBody, FIELD_ID)) {
-    return maxLengthValidation(formBody[FIELD_ID], FIELD_ID, ERROR_MESSAGE, errors, MAXIMUM);
+    return maxLengthValidation(formBody[FIELD_ID], FIELD_ID, ERROR_MESSAGE, errors, MAXIMUM_CHARACTERS.BUYER_COMPANY_OR_ORGANISATION_REGISTRATION_NUMBER);
```
Comment (Zainzzkk) [inline src/ui/server/controllers/insurance/your-buyer/company-or-organisation/validation/rules/registration-number.ts:29]: dereference?

File: src/ui/server/controllers/insurance/your-buyer/credit-insurance-cover/validation/rules/credit-insurance-cover-description.ts [hunk 3]
```diff
@@ -26,11 +23,7 @@ const MAXIMUM = 1000;
  */
 const creditInsuranceCoverDescriptionRule = (formBody: RequestBody, errors: object) => {
   if (formBody[HAS_PREVIOUS_CREDIT_INSURANCE_COVER_WITH_BUYER] === 'true') {
-    if (objectHasProperty(formBody, FIELD_ID)) {
-      return maxLengthValidation(formBody[FIELD_ID], FIELD_ID, ERROR_MESSAGE.ABOVE_MAXIMUM, errors, MAXIMUM);
-    }
-
-    return generateValidationErrors(FIELD_ID, ERROR_MESSAGE.IS_EMPTY, errors);
+    return providedAndMaxLength(formBody, FIELD_ID, ERROR_MESSAGES_OBJECT, errors, MAXIMUM_CHARACTERS.BUYER_PREVIOUS_CREDIT_INSURANCE_COVER_WITH_BUYER);
```
Comment (Zainzzkk) [inline src/ui/server/controllers/insurance/your-buyer/credit-insurance-cover/validation/rules/credit-insurance-cover-description.ts:26]: dereference?

File: src/ui/server/controllers/insurance/business/nature-of-business/validation/rules/goods-or-services.ts (hunk 1 above)
Comment (ttbarnes) [reply src/ui/server/controllers/insurance/business/nature-of-business/validation/rules/goods-or-services.ts:22]: Updated to `BUSINESS.GOODS_OR_SERVICES_DESCRIPTION`

File: src/ui/server/controllers/insurance/your-buyer/company-or-organisation/validation/rules/registration-number.ts (hunk 2 above)
Comment (ttbarnes) [reply src/ui/server/controllers/insurance/your-buyer/company-or-organisation/validation/rules/registration-number.ts:29]: Replaced with the following - I don't think we should dereference in such validation functions, so it's very clear it's a "max characters" definition

```js
BUYER.COMPANY_OR_ORGANISATION
BUYER.REGISTRATION_NUMBER
```

File: src/ui/server/controllers/insurance/your-buyer/credit-insurance-cover/validation/rules/credit-insurance-cover-description.ts (hunk 3 above)
Comment (ttbarnes) [reply src/ui/server/controllers/insurance/your-buyer/credit-insurance-cover/validation/rules/credit-insurance-cover-description.ts:26]: Same as above - update this to be `BUYER.PREVIOUS_CREDIT_INSURANCE_COVER` though.

**Review by Zainzzkk — APPROVED** [review]

**Review by Zainzzkk — APPROVED** [review]

**Pull request merged**